packages/plugin-xml/src/Attributes.ts

```diff
@@ -0,0 +1,45 @@
+import { Modifier } from '../../core/src/Modifier';
+import { deepEqualObjects } from '../../utils/src/utils';
+
+export class Attributes extends Modifier {
+    value: Record<string, string> = {};
+    constructor(valueOrAttributes?: Record<string, string> | Attributes) {
+        super();
+        if (valueOrAttributes instanceof Attributes) {
+            this.value = { ...valueOrAttributes.value };
+        } else if (valueOrAttributes) {
+            this.value = valueOrAttributes;
+        }
+    }
+    get length(): number {
+        return Object.keys(this.value).length;
+    }
+    get name(): string {
+        const name = [];
+        for (const attributeName of Object.keys(this.value)) {
+            name.push(attributeName + ': ' + this.get(attributeName));
+        }
+        return name.join('; ');
+    }
+    get(name: string): string {
+        return this.value[name];
+    }
+    set(name: string, value: string): void {
+        this.value[name] = value;
+    }
+    remove(name: string): void {
+        delete this.value[name];
+    }
+    isSameAs(otherAttributes: Attributes): boolean {
+        if (typeof otherAttributes === 'undefined' && !this.length) return true;
+        return (
+            otherAttributes instanceof this.constructor &&
+            deepEqualObjects(this.value, otherAttributes.value)
```

Comment:
Why deep equal ? It's just a string/string record, shallow key/values comparison should suffice.

Reply:
Right so:
```ts
return (
    otherAttributes instanceof this.constructor &&
    this.length === otherAttributes.length &&
    this.keys().every(key => this._value[key] === otherAttributes._value[key])
);
```

Reply:
You are accessing a private property.